### PR TITLE
Remove specified depths of overlays

### PR DESCRIPTION
**Pull request by Joehuu**
Fixes music controller blocking toolbar input along the way and other fixes from refactor.

Part of #4447.

---

### CODE REVIEW

**Comment by peppy** [comment]
I'm not so sure about the rename to `urgentOverlayContent`..

**Comment by Joehuu** [comment]
`topOverlayContent`, `upperOverlayContent`, `higherOverlayContent`?

**Comment by peppy** [comment]
how about `topMostOverlayContent`? it's a commonly used term for this behaviour

File: osu.Game/OsuGame.cs [hunk 1]
```diff
@@ -399,11 +399,9 @@ protected override void LoadComplete()
...
                     }
                 },
-                overlayContent = new Container
-                {
-                    RelativeSizeAxes = Axes.Both,
-                },
-                floatingOverlayContent = new Container { RelativeSizeAxes = Axes.Both, Depth = float.MinValue },
+                overlayContent = new Container { RelativeSizeAxes = Axes.Both },
+                floatingOverlayContent = new Container { RelativeSizeAxes = Axes.Both },
+                urgentOverlayContent = new Container { RelativeSizeAxes = Axes.Both },
```
Comment (Joehuu) [inline osu.Game/OsuGame.cs]:
```suggestion
                topMostOverlayContent = new Container { RelativeSizeAxes = Axes.Both },
```

File: osu.Game/OsuGame.cs [hunk 2]
```diff
@@ -414,68 +412,49 @@ protected override void LoadComplete()
...
             loadComponentSingleFile(Toolbar = new Toolbar
             {
-                Depth = -5,
                 OnHome = delegate
                 {
                     CloseAllOverlays(false);
                     menuScreen?.MakeCurrent();
                 },
-            }, floatingOverlayContent.Add);
+            }, urgentOverlayContent.Add);
```
Comment (Joehuu) [inline osu.Game/OsuGame.cs]:
```suggestion
            }, topMostOverlayContent.Add);
```

File: osu.Game/OsuGame.cs [hunk 3]
```diff
@@ -414,68 +412,49 @@ protected override void LoadComplete()
...
+                Position = new Vector2(0, Toolbar.HEIGHT),
                 Anchor = Anchor.TopRight,
                 Origin = Anchor.TopRight,
             }, floatingOverlayContent.Add);
 
-            loadComponentSingleFile(accountCreation = new AccountCreationOverlay
-            {
-                Depth = -6,
-            }, floatingOverlayContent.Add);
+            loadComponentSingleFile(accountCreation = new AccountCreationOverlay(), urgentOverlayContent.Add);
```
Comment (Joehuu) [inline osu.Game/OsuGame.cs]:
```suggestion
            loadComponentSingleFile(accountCreation = new AccountCreationOverlay(), topMostOverlayContent.Add);
```

File: osu.Game/OsuGame.cs [hunk 4]
```diff
@@ -414,68 +412,49 @@ protected override void LoadComplete()
...
 
-            loadComponentSingleFile(dialogOverlay = new DialogOverlay
-            {
-                Depth = -7,
-            }, floatingOverlayContent.Add);
+            loadComponentSingleFile(dialogOverlay = new DialogOverlay(), urgentOverlayContent.Add);
```
Comment (Joehuu) [inline osu.Game/OsuGame.cs]:
```suggestion
            loadComponentSingleFile(dialogOverlay = new DialogOverlay(), topMostOverlayContent.Add);
```

File: osu.Game/OsuGame.cs [hunk 5]
```diff
@@ -414,68 +412,49 @@ protected override void LoadComplete()
...
 
-            loadComponentSingleFile(externalLinkOpener = new ExternalLinkOpener
-            {
-                Depth = -8,
-            }, floatingOverlayContent.Add);
+            loadComponentSingleFile(externalLinkOpener = new ExternalLinkOpener(), urgentOverlayContent.Add);
```
Comment (Joehuu) [inline osu.Game/OsuGame.cs]:
```suggestion
            loadComponentSingleFile(externalLinkOpener = new ExternalLinkOpener(), topMostOverlayContent.Add);
```

File: osu.Game/OsuGame.cs [hunk 6]
```diff
@@ -721,6 +700,8 @@ private void updateActiveState(bool isActive)
 
         private Container floatingOverlayContent;
 
+        private Container urgentOverlayContent;
```
Comment (Joehuu) [inline osu.Game/OsuGame.cs]:
```suggestion
        private Container topMostOverlayContent;
```

**Review by Joehuu — COMMENTED** [review]
Using website (not on my computer).

**Review by peppy — APPROVED** [review]
seems ok for now

**Pull request merged**